src/registrar/models/domain.py

```diff
@@ -304,13 +309,40 @@ def isSubdomain(cls, name: str, nameserver: str):
         regex = re.compile(full_pattern)
         return bool(regex.match(nameserver))
 
+    @classmethod
+    def isValidDomain(cls, nameserver: str):
```

Comment:
**question (non-blocking)**: It's not very easy to find, but we already have a `Domain.string_could_be_domain()` method with similar regexes in it. Can we clarify somehow if there is a difference between how this and that behave and where we should be using each one?

Reply:
There was some redundant code, which became obsolete in previous iterations.  I removed redundant code and obsolete code.  Part of the reason for doing this, rather than refactoring to the older model is that (1) the older model was processing all tuples at once and not one form at a time, and (2)  I am anticipating that there could be a future desire for different NameserverErrors and error messages for users to be raised for different conditions, and would like these exceptions to be raised from the same area in the code for code readability.